### PR TITLE
LPS-94581 Remove user customizations when customizable pages are deleted

### PR DESCRIPTION
**Pull request by jrao**
https://issues.liferay.com/browse/LPS-94581 Remove user customizations when customizable pages are deleted

### CODE REVIEW

**Comment by jrao** [comment]
ci:test:sf

**Comment by jrao** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java [hunk 1]
```diff
@@ -840,6 +844,14 @@ public void deleteLayout(
 
 		layout = delete(layout);
 
+		// User preferences for customizable pages
+
+		if (layout.isCustomizable()) {
```
Comment (jrao) [inline portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java]: @ChrisKian - something to think about: maybe this check is unnecessary? With this check, user preferences would not be deleted if a page was modified to be not customizable before it was deleted. And looking at this again, I don't see any downside to running the _removeUserPreferences method for all pages (besides performance considerations).

**Comment by ChrisKian** [comment]
Hey @jrao,
I have a few things I'd like you to look through and change if needed:

1. Good point on your above comment regarding the necessary check.  Could you please do a bit of testing to see if it's required or not?  If a page can be modified to non-customizable, I would check to see if the preferences should be removed, if they are removed, and lastly if they aren't removed, where should they be removed (should probably be removed when the page is changed to non-customizable).

2. Performing a [getPortalPreferences() on all entries](https://github.com/ChrisKian/liferay-portal/pull/134/files#diff-fd3afdf01e0bd16b0bd10406ac6952adR3598) is a bad move.  We always want to return as few entries as possible from the DB.  See if there is a different method which can filter through the entries needed (maybe lookup using layout or plid?).

3. I don't like how you make a portletPreferencesImpl object to update the value.  Just get the preferences, search for the value, remove it, portalPreferences.setPreferences(), then update using PortalPreferencesLocalServiceUtil.

Let me know if you have any questions, thanks!

**Comment by jrao** [comment]
Sent https://github.com/ChrisKian/liferay-portal/pull/137

**Pull request closed without merging**